apollo-router/src/plugins/authorization/scopes.rs

```diff
@@ -26,23 +26,43 @@ pub(crate) struct ScopeExtractionVisitor<'a> {
     compiler: &'a ApolloCompiler,
     file_id: FileId,
     pub(crate) extracted_scopes: HashSet<String>,
+    requires_scopes_directive_name: String,
 }
 
 pub(crate) const REQUIRES_SCOPES_DIRECTIVE_NAME: &str = "requiresScopes";
 
 impl<'a> ScopeExtractionVisitor<'a> {
     #[allow(dead_code)]
-    pub(crate) fn new(compiler: &'a ApolloCompiler, file_id: FileId) -> Self {
-        Self {
+    pub(crate) fn new(compiler: &'a ApolloCompiler, file_id: FileId) -> Option<Self> {
+        let requires_scopes_directive_name = if let Some(link) = compiler
+            .db
+            .schema()
+            .directives_by_name("link")
+            .filter(|link| {
+                link.argument_by_name("url")
+                    .and_then(|value| value.as_str())
+                    == Some("https://specs.apollo.dev/requiresScopes/v0.1")
+            })
+            .next()
+        {
+            link.argument_by_name("as")
+                .and_then(|value| value.as_str().map(|s| s.to_string()))
+                .unwrap_or_else(|| REQUIRES_SCOPES_DIRECTIVE_NAME.to_string())
+        } else {
+            return None;
+        };
```

Comment:
This should be a function to not duplicate that part of code and risk a misspelled name

Reply:
or create constants for identifiers

Reply:
moved to https://github.com/apollographql/router/pull/3949